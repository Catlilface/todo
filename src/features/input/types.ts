export default interface IInputProps {
  defaultValue?: string;
  onEnter?: CallableFunction;
  className?: string;
  placeholder?: string;
}
