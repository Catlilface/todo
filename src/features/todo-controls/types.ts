import { TodoMode } from "../../widgets/todo/types";

export interface ITodoControls {
  onChange: CallableFunction;
  mode: TodoMode;
}
