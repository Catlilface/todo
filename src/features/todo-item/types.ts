import { ITodoItem } from "../../widgets/todo/types";

export interface ITodoItemFeature extends ITodoItem {
  onCheckTodo: CallableFunction;
}
