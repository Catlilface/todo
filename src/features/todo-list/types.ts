import { TodoMode } from "../../widgets/todo/types";

export interface ITodoList {
  mode: TodoMode;
  onCheckTodo: CallableFunction;
}
