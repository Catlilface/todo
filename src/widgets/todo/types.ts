export interface ITodoItem {
  id: string;
  title: string;
  completed: boolean;
}

export default interface ITodo {
  todos: ITodoItem[];
}

export enum TodoMode {
  ALL = "all",
  COMPLETED = "completed",
  ACTIVE = "active",
}
