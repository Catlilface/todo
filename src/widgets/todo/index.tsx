import Input from "../../features/input";
import { checkTodoFn, createTodoFn } from "../../store";
import { useUnit } from "effector-react";
import { TodoMode } from "./types";
import { useState } from "react";
import { TodoControls, TodoList } from "../../features";

export default function Todo() {
  const createTodo = useUnit(createTodoFn);
  const checkTodo = useUnit(checkTodoFn);
  const [mode, setMode] = useState<TodoMode>(TodoMode.ALL);

  return (
    <div className="bg-zinc-900 rounded-3xl woww-so-modern w-full max-w-[400px] mt-8">
      <Input
        className="py-3 rounded-md w-full bg-transparent focus-visible:outline-none text-center border-b-2 border-zinc-800"
        onEnter={createTodo}
        placeholder="Хмм, сегодня мне надо..."
      />
      <TodoList mode={mode} onCheckTodo={checkTodo} />
      <TodoControls mode={mode} onChange={setMode} />
    </div>
  );
}
