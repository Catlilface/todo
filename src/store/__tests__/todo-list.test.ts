import { describe, expect, test } from "vitest";
import { TodoMode } from "../../widgets/todo/types";
import { getTodos } from "..";

describe("Фильтрация с todo", async () => {
  for (const mode of Object.values(TodoMode)) {
    test(`Фильтрация с todo ${mode}`, async () => {
      const todos = getTodos(mode);

      expect(todos.defaultState).toMatchSnapshot();
    });
  }
});
