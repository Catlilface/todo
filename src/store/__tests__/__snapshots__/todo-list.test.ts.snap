// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Фильтрация с todo > Фильтрация с todo active 1`] = `
[
  {
    "completed": false,
    "id": "todo-1",
    "title": "Купить хлеб",
  },
  {
    "completed": false,
    "id": "todo-2",
    "title": "Купить молоко",
  },
  {
    "completed": false,
    "id": "todo-4",
    "title": "Посмотреть кино",
  },
]
`;

exports[`Фильтрация с todo > Фильтрация с todo all 1`] = `
[
  {
    "completed": false,
    "id": "todo-1",
    "title": "Купить хлеб",
  },
  {
    "completed": false,
    "id": "todo-2",
    "title": "Купить молоко",
  },
  {
    "completed": true,
    "id": "todo-3",
    "title": "Пройти собеседование",
  },
  {
    "completed": false,
    "id": "todo-4",
    "title": "Посмотреть кино",
  },
  {
    "completed": true,
    "id": "todo-5",
    "title": "Посмотреть в окно",
  },
  {
    "completed": true,
    "id": "todo-6",
    "title": "Попасть в просак",
  },
]
`;

exports[`Фильтрация с todo > Фильтрация с todo completed 1`] = `
[
  {
    "completed": true,
    "id": "todo-3",
    "title": "Пройти собеседование",
  },
  {
    "completed": true,
    "id": "todo-5",
    "title": "Посмотреть в окно",
  },
  {
    "completed": true,
    "id": "todo-6",
    "title": "Попасть в просак",
  },
]
`;
